silverback/types.py

```diff
@@ -44,14 +44,17 @@ class _BaseDatapoint(BaseModel):
 
 
 # NOTE: Maximum supported parquet integer type: https://parquet.apache.org/docs/file-format/types
-Int96 = Annotated[int, Field(ge=-(2**95), le=2**95 - 1)]
+INT96_RANGE = (-(2**95), 2**95 - 1)
+Int96 = Annotated[int, Field(ge=INT96_RANGE[0], le=INT96_RANGE[1])]
 # NOTE: only these types of data are implicitly converted e.g. `{"something": 1, "else": 0.001}`
-ScalarType = bool | Int96 | float | Decimal
+PydanticScalarType = bool | Int96 | float | Decimal
```

Comment:
I actually found a way to get the wrapped class of annotated tags, use `.__origin__`

Reply:
I'm not really sure what you're suggesting.  Link or example?

Reply:
`get_args` gives you `typing._AnnotatedType` or something, which has `.__origin__` that shows you that the inner wrapped type is `int`

Reply:
Not trying to be dense but I'm not really following how that would be useful here.  Feel free to push any changes.

Reply:
I did this, which can be used directly in the `isinstance` check:
```py
SCALAR_BASE_TYPES = tuple(t.__origin__ if hasattr(t, "__origin__") else t for t in get_args(ScalarType))
```

Reply:
can also do something like this instead of `isinstance`:

```py
def is_scalar(val: Any) -> bool:
    return isinstance(SCALAR_BASE_TYPES)
```

seems even clearer